src/desc.rs

```diff
@@ -5,32 +5,76 @@ use std::collections::{BTreeSet, HashMap};
 use std::hash::Hasher;
 
 use fnv::FnvHasher;
-use regex::Regex;
 
 use crate::errors::{Error, Result};
 use crate::metrics::SEPARATOR_BYTE;
 use crate::proto::LabelPair;
 
-// Details of required format are at
-// https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
-fn is_valid_metric_name(name: &str) -> bool {
-    lazy_static! {
-        static ref VALIDATOR: Regex =
-            Regex::new("^[a-zA-Z_:][a-zA-Z0-9_:]*$").expect("Regex to be valid.");
+#[cfg(not(feature = "regex"))]
+mod validation {
+    fn matches_charset_without_colon(c: char) -> bool {
```

Comment:
```suggestion
    fn matches_alphabetic(c: char) -> bool {
```

Reply:
It's not just alphabetic, it's also underscore. Do you want that name anyway?

Reply:
True, sorry. Not a good suggestion from my side.